### PR TITLE
Build ROS packages on travis

### LINKED ISSUE #386: catkin_make on Travis
We should build all our packages on travis to ensure the build is correct.

We might want to look into an image with ros preinstalled to avoid making Travis install 30 minutes of shit.

https://github.com/felixduvallet/ros-travis-integration

### PR DESCRIPTION
**Pull request by AFknUsername**
# Assignee Section

## Description
Travis now builds our ROS packages

### Steps for Testing
Check the travis logs I guess?

closes #386 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [ ] Local Test Performed Successfully
- [ ] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
 language: python
+sudo: required
+cache:
+  - apt
+env:
+  global:
+    # Adapted from https://github.com/felixduvallet/ros-travis-integration/blob/master/.travis.yml
+    - CI_SOURCE_PATH=$(pwd)
+    - ROS_PARALLEL_JOBS='-j8 -l6'
+    - PYTHONPATH=$PYTHONPATH:$CI_SOURCE_PATH/venv/lib/python3.6/dis-packages
```
Comment (MartensCedric) [inline .travis.yml:12]: Given that you added new environment variables, you should look around and make sure the code is consistent. The 3 lines below makes this inconsistent. When I added them, I was not aware of the env global setting, so maybe you should moove all these "export VARIABLE" up here.

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,15 @@
 os: linux
-dist: trusty
+dist: xenial
 language: python
+sudo: required
```
Comment (MartensCedric) [inline .travis.yml:4]: image

Pretty sure the sudo line can be safely removed

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
+cache:
+  - apt
+env:
+  global:
+    # Adapted from https://github.com/felixduvallet/ros-travis-integration/blob/master/.travis.yml
+    - CI_SOURCE_PATH=$(pwd)
```
Comment (PeterGhimself) [inline .travis.yml:10]: Is `CI_SOURCE_PATH` just the root of the repo? Can I suggest to change it to `REPO_ROOT` in that case?

File: .travis.yml [hunk 4]
```diff
@@ -16,12 +25,30 @@ before_install:
...
+  - cd .. && catkin_make
+  - source devel/setup.bash
+before_script:
+  - rosdep install -y -r --from-paths src --ignore-src
 script:
-  - cd robot/rover/build
+  - cd $CI_SOURCE_PATH/robot/rover/build
   - cmake ..
   - make
   - cd ../../..
```
Comment (PeterGhimself) [inline .travis.yml:54]: Assuming `CI_SOURCE_PATH` is repo root, we can change this to `cd $CI_SOURCE_PATH`, or `cd $ROOT_REPO` if you agree to rename the variable

File: .travis.yml (hunk 1 above)
Comment (AFknUsername) [reply .travis.yml]: done

File: .travis.yml (hunk 3 above)
Comment (AFknUsername) [reply .travis.yml]: done

File: .travis.yml (hunk 2 above)
Comment (AFknUsername) [reply .travis.yml]: removec

File: .travis.yml (hunk 4 above)
Comment (AFknUsername) [reply .travis.yml]: changed

**Pull request merged**